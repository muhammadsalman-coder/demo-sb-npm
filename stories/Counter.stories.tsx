import React from 'react';
import { Meta, Story } from '@storybook/react';
import { SalmanCounter, SalmanPropsInterface } from '../src/SalmanCounter';
const meta: Meta = {
  title: 'Counter',
  component: SalmanCounter,
  argTypes: {
    onchange: { action: 'onchange' },
  },
};
// argTypes: {
//   children: {
//     control: {
//       type: 'text',
//     },
//   },
// },
// parameters: {
//   controls: { expanded: true },
// },

export default meta;

const Template: Story<SalmanPropsInterface> = args => (
  <SalmanCounter {...args} />
);

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({
  children: <></>,
});

Default.args = {
  yourName: 'Shan',
};
