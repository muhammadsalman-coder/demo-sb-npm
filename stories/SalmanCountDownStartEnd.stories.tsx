import React from 'react';
import { Meta, Story } from '@storybook/react';
import { SalmanCountDownStartEnd } from '../src/SalmanCountDownStartEnd';

const meta: Meta = {
  title: 'SalmanCountDownStartEnd',
  component: SalmanCountDownStartEnd,
  argTypes: {
    endDateTime: { control: 'date' },
    startDateTime: { control: 'date' },
    color: { control: 'color' },
  },
};
// argTypes: {
//   children: {
//     control: {
//       type: 'text',
//     },
//   },
// },
// parameters: {
//   controls: { expanded: true },
// },

export default meta;

const Template: Story<any> = args => <SalmanCountDownStartEnd {...args} />;

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});

Default.args = {
  timerEndText: 'ended Timer',
  timerStartText: 'Start on: ',
  gap: 30,
  color: '#000',
  timeHeadinSize: '26px',
  timeTextSize: '16px',
  timerEndSize: '16px',
};
