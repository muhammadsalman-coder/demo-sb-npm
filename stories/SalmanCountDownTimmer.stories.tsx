import React from 'react';
import { Meta, Story } from '@storybook/react';
import { SalmanCountDownTimmer } from '../src/SalmanCountDownTimmer';

const meta: Meta = {
  title: 'SalmanCountDownTimmer',
  component: SalmanCountDownTimmer,
  argTypes: {
    endDateTime: { control: 'date' },
    startDateTime: { control: 'date' },
    color: { control: 'color' },
  },
};
// argTypes: {
//   children: {
//     control: {
//       type: 'text',
//     },
//   },
// },
// parameters: {
//   controls: { expanded: true },
// },

export default meta;

const Template: Story<any> = args => <SalmanCountDownTimmer {...args} />;

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});

Default.args = {
  timerEndText: 'ended Timer',
  gap: 30,
  color: '#000',
  timeHeadinSize: '26px',
  timeTextSize: '16px',
  timerStartText: 'Timer Start on: ',
  timerStartSize: '16px',
  timerEndSize: '16px',
};
