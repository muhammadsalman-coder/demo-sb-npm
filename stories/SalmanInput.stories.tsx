import React from 'react';
import { Meta, Story } from '@storybook/react';
import { SalmanInput, SalmanInputInterface } from '../src/SalmanInput';
const meta: Meta = {
  title: 'Input',
  component: SalmanInput,
};
// argTypes: {
//   children: {
//     control: {
//       type: 'text',
//     },
//   },
// },
// parameters: {
//   controls: { expanded: true },
// },

export default meta;

const Template: Story<SalmanInputInterface | any> = args => (
  <SalmanInput {...args} onChange={console.log} />
);

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});
Default.args = {
  borderRadius: '20px',
  type: 'text',
  placeholder: 'enter name',
  padding: '10px',
  width: '100%',
};
export const TextInput = Template.bind({});
TextInput.args = {
  borderRadius: '20px',
  type: 'text',
  placeholder: 'enter name',
  padding: '10px',
  width: '100%',
};
export const PasswordInput = Template.bind({});
PasswordInput.args = {
  borderRadius: '20px',
  type: 'password',
  placeholder: 'enter password',
  padding: '10px',
  width: '100%',
};
export const FileInput = Template.bind({});
FileInput.args = {
  borderRadius: '20px',
  type: 'file',
  placeholder: 'enter password',
  padding: '10px 10px 10px 35px',
  width: 'fit-content',
};
