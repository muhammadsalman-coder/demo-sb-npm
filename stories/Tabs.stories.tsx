import React from 'react';
import { Meta, Story } from '@storybook/react';
import { SalmanTabs, ItemInterface } from '../src/SalmanTabs';
import HomeIcon from '@mui/icons-material/Home';
const meta: Meta = {
  title: 'Tabs',
  component: SalmanTabs,
};
// argTypes: {
//   children: {
//     control: {
//       type: 'text',
//     },
//   },
// },
// parameters: {
//   controls: { expanded: true },
// },

export default meta;

const Template: Story<ItemInterface> = args => (
  <SalmanTabs
    item={[
      { text: 'Home', icon: <HomeIcon /> },
      { text: 'About', icon: <HomeIcon /> },
      { text: 'Contact', icon: <HomeIcon /> },
      { text: 'Home', icon: <HomeIcon /> },
      { text: 'About', icon: <HomeIcon /> },
      { text: 'Contact', icon: <HomeIcon /> },
      { text: 'Home', icon: <HomeIcon /> },
      { text: 'About', icon: <HomeIcon /> },
      { text: 'Contact', icon: <HomeIcon /> },
      { text: 'Home', icon: <HomeIcon /> },
      { text: 'About', icon: <HomeIcon /> },
      { text: 'Contact', icon: <HomeIcon /> },
    ]}
    {...args}
  />
);

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});

Default.args = {
  iconSize: '16px',
  fontSize: '18px',
  maxWidth: '500px',
  activeUnderline: true,
  activecolor: '#353840',
  iconColor: '#707a83',
  color: '#707a83',
  scrollThumbColor: '#707a83',
  item: [
    { text: 'Home', icon: <HomeIcon /> },
    { text: 'About', icon: <HomeIcon /> },
    { text: 'Contact', icon: <HomeIcon /> },
    { text: 'Home', icon: <HomeIcon /> },
    { text: 'About', icon: <HomeIcon /> },
    { text: 'Contact', icon: <HomeIcon /> },
    { text: 'Home', icon: <HomeIcon /> },
    { text: 'About', icon: <HomeIcon /> },
    { text: 'Contact', icon: <HomeIcon /> },
    { text: 'Home', icon: <HomeIcon /> },
    { text: 'About', icon: <HomeIcon /> },
    { text: 'Contact', icon: <HomeIcon /> },
  ],
};
