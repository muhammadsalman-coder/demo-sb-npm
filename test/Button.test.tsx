import React from 'react';
import { Default as Button } from '../stories/Button.stories';
import { render } from '@testing-library/react';

describe('Button test', () => {
  it('renders', () => {
    const { baseElement } = render(
      <Button onChange={console.log} color="primary" backgroundColor="red">
        Bitspro
      </Button>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
