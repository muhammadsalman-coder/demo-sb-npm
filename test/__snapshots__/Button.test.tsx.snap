// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test renders 1`] = `
<body>
  <div>
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1435yey-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Bitspro
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;
